src/lib/mina/transaction.ts

```diff
@@ -280,13 +280,32 @@ type RejectedTransaction = Pick<
   errors: string[];
 };
 
+/**
+ * A `Promise<Transaction>` with some additional methods for making chained method calls
+ * into the pending value upon its resolution.
+ */
 type TransactionPromise = Promise<Transaction> & {
+  /** Equivalent to calling the resolved `Transaction`'s `sign` method. */
   sign(...args: Parameters<Transaction['sign']>): TransactionPromise;
+  /** Equivalent to calling the resolved `Transaction`'s `send` method. */
   send(): PendingTransactionPromise;
+  /**
+   * Calls `prove` upon resolution of the `Transaction`. Returns a
+   * new `TransactionPromise` with the field `proofPromise` containing
+   * a promise which resolves to the proof array.
+   */
+  prove(): TransactionPromise;
+  /**
+   * If the chain of method calls that produced the current `TransactionPromise`
+   * contains a `prove` call, then this field contains a promise resolving to the
+   * proof array which was output from the underlying `prove` call.
+   */
+  proofPromise?: Promise<(Proof<ZkappPublicInput, undefined> | undefined)[]>;
```

Comment:
slightly suboptimal IMO that this makes the the return of `prove()` only optionally have the `proofPromise` property, maybe you can tweak the return type of `prove()`?

Reply:
Just updated the `TransactionPromise` type such that it has a type param `Proven extends boolean`, which defaults to `false`. This type param determines whether the developer sees the `prove` method or the `proofPromise` member. Calling `prove` returns a `TransactionPromise<false>`.